modules/otel-config-target-allocator.adoc

```diff
@@ -0,0 +1,93 @@
+// Module included in the following assemblies:
+//
+// * otel/otel-configuration-of-collector.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="otel-config-target-allocator_{context}"]
+= Target allocator
+
+The target allocator is an optional component of the OpenTelemetry Operator that shards scrape targets across the deployed fleet of OpenTelemetry Collector instances. The target allocator integrates with the Prometheus `PodMonitor` and `ServiceMonitor` custom resources. When the target allocator is enabled,  the OpenTelemetry Operator adds `http_sd_config` to the enabled `prometheus` receiver that connects to the target allocator service.
```

Comment:
```suggestion
The target allocator is an optional component of the OpenTelemetry Operator that shards scrape targets across the deployed fleet of OpenTelemetry Collector instances. The target allocator integrates with the Prometheus `PodMonitor` and `ServiceMonitor` custom resources (CR). When the target allocator is enabled,  the OpenTelemetry Operator adds `http_sd_config` to the enabled `prometheus` receiver that connects to the target allocator service.
```

define CR here since you use it in the example heading below

Reply:
Thank you. I'm applying your suggestion through my editor (to keep just one commit here).